src/prefect/tasks/core/function.py

```diff
@@ -10,9 +10,20 @@
 import prefect
 
 
+class _DocProxy(object):
+    def __init__(self, source):
+        self._source = source
+
+    def __get__(self, obj, cls):
+        if obj is None:
+            return self._source
+        else:
+            return obj.run.__doc__
+
+
 class FunctionTask(prefect.Task):
-    """
-    A convenience Task for functionally creating Task instances with
+    __doc__ = _DocProxy(
```

Comment:
Could you explain this `DocProxy` object?  I haven't seen this before and don't fully get the idea

Reply:
Sure. This is a descriptor object ([this](https://docs.python.org/3/howto/descriptor.html) is a pretty good ref) used to make the `__doc__` attribute different between the class `FunctionTask` and for an instance of `FunctionTask`.

- `FunctionTask.__doc__` is the docstring here
- `FunctionTask(...).__doc__` is the docstring of the wrapped function

We could also assign `self.__doc__` in the constructor, but that will duplicate the docstring in serialization (in a future PR I also want to cleanup how `FunctionTask`s serialize, as currently they aren't picklable for *reasons*). That said, this is mostly leftover from playing around with different locations for docstrings and how ipython and sphinx load those (I tried overriding the `__call__` docstring first, which required a descriptor to handle, the class docstring doesn't).

Reply:
Ah very cool - TIL!